subprojects/gradle/build-metrics/src/gradleTest/kotlin/com/avito/android/plugin/build_metrics/cache/BuildCacheMetricsTest.kt

```diff
@@ -0,0 +1,176 @@
+package com.avito.android.plugin.build_metrics.cache
+
+import com.avito.android.plugin.build_metrics.assertHasMetric
+import com.avito.android.stats.CountMetric
+import com.avito.test.gradle.TestResult
+import com.google.common.truth.Truth.assertThat
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.jupiter.api.Test
+import java.io.File
+
+internal class BuildCacheMetricsTest : BuildCacheTestFixture() {
```

Comment:
Integration test for new metrics are here

Reply:
Brilliant tests, btw! I really like how it's structured!
Only possible improvement i see is a assertion by outgoing http requests to stub statsd instead of logs